src/python_routing_v02/troute/routing/compute.py

```diff
@@ -71,18 +72,20 @@ def _prep_da_dataframes(
     time series is as long as the simulation.
 
     """
+    # segments in the subnetwork ONLY, no offnetwork upstreams included
+    subnet_segs = param_df_sub_idx.symmetric_difference(offnetwork_upstreams)
+    
     # NOTE: Uncomment to easily test no observations...
     # usgs_df = pd.DataFrame()
     if not usgs_df.empty and not lastobs_df.empty:
         # index values for last obs are not correct, but line up correctly with usgs values. Switched
-
-        lastobs_segs = list(lastobs_df.index.intersection(param_df_sub_idx))
+        lastobs_segs = list(lastobs_df.index.intersection(subnet_segs))
         lastobs_df_sub = lastobs_df.loc[lastobs_segs]
-        usgs_segs = list(usgs_df.index.intersection(param_df_sub_idx))
+        usgs_segs = list(usgs_df.index.intersection(subnet_segs))
         da_positions_list_byseg = param_df_sub_idx.get_indexer(usgs_segs)
```

Comment:
This line remains unchanged, ensuring that the `da_positions_list_byseg` is correctly indexing `param_df_sub_idx`